apps/workspace-host/src/interface.js

```diff
@@ -920,11 +920,6 @@ async function initSequenceAsync(workspace_id, useInitialZip, res) {
 
     try {
       await workspaceUtils.updateWorkspaceMessage(workspace.id, 'Creating container');
-      const hostname = `${workspace_server_settings.server_to_container_hostname}:${workspace.launch_port}`;
-      await sqldb.queryAsync(sql.update_workspace_hostname, {
-        workspace_id,
-        hostname,
-      });
```

Comment:
The fact that we blindly write the `hostname` potentially a long time after the launch request starts is the root problem here. In theory, another solution would be to write the hostname at the same time that we write the `launch_uuid`; I think things would behave the same in the end. We'd still have to check the `launch_uuid` before transitioning to the running state, however.